### ISSUE TITLE
Top down / bottom up - Brell gateway

### ISSUE THREAD

**Issue by taugusti** [starkillerOG/motion-blinds#29]
Thanks for providing this integration.

I have an issue with the integration or brel gateway on my top-down / bottom-up curtains. They have two motors and the top and bottom motors show up as separate entities in home asssistant (top, bottom and combined).

Cover open and close works fine - but when I try to set the position (set position) it only works with the bottom motor.
The same goes for the motion_blinds.set_absolute_position service.

How can I debug this to get some indication on where the problem is?
Thanks

**Comment by starkillerOG** [comment]
@taugusti did you try the user interface of the top motor?
Does the combined entity work through the user interface?

Do you see any errors in the homeassistant log about the motion blinds integration, specifically about the postion not beeing allowed (due to collision of the top en bottom bar)?

Note that since the top motor can not go below the bottom motor there are absolute positions that are not allowed, depending on the position of the bottom motor (making these TDBU motors a bit difficult).

Have you already read the documentation section about the TDBU motors: https://www.home-assistant.io/integrations/motion_blinds/#top-down-bottom-up-tdbu-blinds?
That gives some explanation on how the diffrent properties work.

**Comment by starkillerOG** [comment]
If you appreciate the motion blinds integration, please consider sponsoring this library: https://github.com/sponsors/starkillerOG

**Comment by taugusti** [comment]
Hi @starkillerOG 
Thanks for fast reply

1. Does it work through the UI?
Partially... I can pull the slider for the top motor and it (sometimes) starts to move. but it does not stop at the position I request and sometimes it just refuses to move at all. if the motor is in motion, (also if I use the up/down bottons), I am unable to stop the motion using the stop button.
The log shows: "Send status that did not include the position of the TDBU"
I also noted that the "absolute position" attribute in the UI does not update when the blind moves.

For reference I did the same with a bottom motor and I get the same error in the log and absolute position is not update - but it stops at the position I set it to.

2. logged errors when setting position
Yes there are errors - should have thought to check the log and reported in the in first report - sorry about that.
I get two errors what ever I do:
"Send status that did not include the position of the TDBU"
"Error setting position, the top of the TDBU can not go below the bottom of theTDBU blind"

I have tried all combinations of position / absolute position with / without width. even tried negative positions. Nothing seems to work.

Thanks for your help

**Comment by starkillerOG** [comment]
@taugusti alright so the problem seems to be that the TDBU is not including the position of the motors in its status updates. For TDBU's this is a real problem since we need to know the current positions to determine if a new position is allowed or not.

How does the brel app work?
Does it acurately provide the position of the TDBU?
Can you move both TDBU motors in the brel app and can you see the positions chaning in the app?

I would advice to use the `motion_blinds.set_absolute_position` with the **combined** entity_id and then spicify both `absolute_position` and `width`. You then do not need to know the current position of the motors and schould be able to move the TDBU consistently.
Can you verify if this works?

**Comment by taugusti** [comment]
@starkillerOG 
**Brel app**
Only one of 4 blinds report their position in the app.
I can set the position of the other 3 blinds - both top and bottom - bot their position is not reported in the app.
I will contact brel with this issue - perhaps fixing it will help on the hass integration as well.

**combined entity**
It seems to work - great!
It took me a little while to realize that the absolute position is right between the top and bottom... :-)
For the automations / scenes I am planning to use right now, the combined entry will work ok.

I will leave it open to report back on what Brel says.

**Comment by taugusti** [comment]
Hi @starkillerOG 
Finally got the cae solved. 
Not much feedback from Brell - but by accident I moved the gateway a little bit - and one of the blinds stopped responding!
So... I realized that the gateway was too far from the curtains. When I moved it closer, it worked again. 
AND the position feedback suddenly worked also.
Conclusion: nothing wrong anywhere and the advice for future users - be careful with the range :-)
Thanks for your support
I will close the issue.
